django_emarsys/checks.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import logging
+import re
+
+from django.apps import apps
+from django.conf import settings
+from django.core.checks import register, Critical, Debug, Error, Info, Warning
+
+log = logging.getLogger(__file__)
+
+
+@register()
+def check_settings(app_configs):
+    return validate_settings()
+
+
+def add_message(messages, message):
+    messages.append(message)
+    log_message = {Debug: log.debug,
+                   Info: log.info,
+                   Warning: log.warning,
+                   Error: log.error,
+                   Critical: log.critical}[type(message)]
+
+    log_message(message.msg)
+
+
+def validate_settings():
+    """
+    Turn settings into a configuration dict that `validate_configuration` can
+    check.
+
+    The extra `hasattr()` check is necessary to make this work with overridden
+    settings from tests. If a setting existed in the original settings object,
+    but is then deleted in overridden settings, then `dir()` will report it
+    present, but accessing it raises an `AttributeError`.
+
+    Minimal test for this:
+
+    >>> from django.conf import settings
+    >>> from django.test.utils import override_settings
+    >>> @override_settings()
+    >>> def test_strange_dir_issue():
+    >>>     delattr(settings, 'DEBUG')
+    >>>     assert 'DEBUG' in dir(settings)
+    >>>     assert not hasattr(settings, 'DEBUG')
+    >>>
+    >>> test_strange_dir_issue()
+
+    """
+    return validate_configuration({key: getattr(settings, key)
+                                  for key in dir(settings)
+                                  if hasattr(settings, key)})
+
+
+def validate_configuration(config):
+    messages = []
+    if not config.get('EMARSYS_ACCOUNT'):
+        add_message(messages, Critical("EMARSYS_ACCOUNT not set"))
```

Comment:
I'd rather have a variable be modified in one place, than in two. As it stands, add_message has side effects.

Since this point is moot if we decide not to log the messages I'll wait until we make that decision.